rootex/core/resource_files/basic_material_resource_file.h

```diff
@@ -45,6 +45,8 @@ class BasicMaterialResourceFile : public MaterialResourceFile
 	void bindVSCB() override;
 	void bindPSCB() override;
 
+	Ref<ImageResourceFile> getDiffuseImageFile() { return m_DiffuseImageFile; }
```

Comment:
no need for getter, access like other textures have been accessed, also doesn't make sense to return a Ref, since the object which calls this function will most probably not hold ownership of this.

Reply:
How to do it without a getter?